Remove terraform state management and unique s3 bucket name (#834)
openfun/marsha: README.md

All tasks related to this environment are run from the `./src/aws` directory. We use `Terraform` to keep this infrastructure configuration as code and easily manage several independent deployments of the whole `AWS` infrastructure.

> There are 2 Terraform projects in Marsha with two differents purpose:

Reviewer: ```suggestion
There are 2 Terraform projects in Marsha with two different purposes:
```

> - `./src/aws/shared_resources`: this project managed resources common to all marsha instances. These resources must not live in different workspaces and you must work in the `default` workspace. To ease the use of this project, a dedicated script is available in `./src/aws/bin/shared-resources`, it uses and configures the `Terraform` docker image, you just have to run a terraform command like if you were using the terraform cli. (eg: `./bin/shared-resources plan` will generate the execution plan of terraform).

Reviewer: ```suggestion
- `./src/aws/shared_resources`: this project manages resources common to all marsha environments. These resources need not live in different workspaces so you must work in the `default` workspace. To ease the use of this project, a dedicated script is available in `./src/aws/bin/shared-resources` which uses and configures the `Terraform` docker image. You have to run a Terraform command as if you were using the terraform cli. (eg: `./bin/shared-resources plan` will execute Terraform's "plan" command).
```

> - `./src/aws/shared_resources`: this project managed resources common to all marsha instances. These resources must not live in different workspaces and you must work in the `default` workspace. To ease the use of this project, a dedicated script is available in `./src/aws/bin/shared-resources`, it uses and configures the `Terraform` docker image, you just have to run a terraform command like if you were using the terraform cli. (eg: `./bin/shared-resources plan` will generate the execution plan of terraform).

Author: It's about the instances on the same aws account.

> - `./src/aws/shared_resources`: this project managed resources common to all marsha instances. These resources must not live in different workspaces and you must work in the `default` workspace. To ease the use of this project, a dedicated script is available in `./src/aws/bin/shared-resources`, it uses and configures the `Terraform` docker image, you just have to run a terraform command like if you were using the terraform cli. (eg: `./bin/shared-resources plan` will generate the execution plan of terraform).

Author: So **mine** instances

> - `./src/aws/shared_resources`: this project managed resources common to all marsha instances. These resources must not live in different workspaces and you must work in the `default` workspace. To ease the use of this project, a dedicated script is available in `./src/aws/bin/shared-resources`, it uses and configures the `Terraform` docker image, you just have to run a terraform command like if you were using the terraform cli. (eg: `./bin/shared-resources plan` will generate the execution plan of terraform).

Reviewer 2: :+1:  I would suggest the following :

```suggestion
- `./src/aws/shared_resources`: this project managed resources common to all marsha instances on the same AWS account. These resources must not live in different workspaces and you must work in the `default` workspace. To ease the use of this project, a dedicated script is available in `./src/aws/bin/shared-resources`, it uses and configures the `Terraform` docker image, you just have to run a terraform command like if you were using the terraform cli. (eg: `./bin/shared-resources plan` will generate the execution plan of terraform).
```

> - `./src/aws/shared_resources`: this project managed resources common to all marsha instances. These resources must not live in different workspaces and you must work in the `default` workspace. To ease the use of this project, a dedicated script is available in `./src/aws/bin/shared-resources`, it uses and configures the `Terraform` docker image, you just have to run a terraform command like if you were using the terraform cli. (eg: `./bin/shared-resources plan` will generate the execution plan of terraform).

Reviewer 2: Indeed, it is more clear :+1:

> - `./src/aws`: this is the main project we use, most of the infrastructure is managed here (in all `*.tf` files). This project must use [`Terraform` workspaces](https://www.terraform.io/docs/state/workspaces.html) and we highly recommand you to not use the default one. With multiple workspaces you can manage multiple instances of Marsha with one AWS account. To ease the use of this project, a dedicated script is available in `./src/aws/bin/terraform`, it uses and configures the `Terraform` docker image, you just have to run a terraform command like if you were using the terraform cli. (eg: `./bin/terraform plan` will generate the execution plan of terraform).

Reviewer: ```suggestion
- `./src/aws`: this is the main project we use, most of the infrastructure is managed here (in all `*.tf` files). This project must use [`Terraform` workspaces](https://www.terraform.io/docs/state/workspaces.html) and we highly recommand you to not use the default one. With multiple workspaces you can manage multiple instances of Marsha with one AWS account. To ease the use of this project, a dedicated script is available in `./src/aws/bin/terraform` which uses and configures the `Terraform` docker image. You have to run a Terraform command as if you were using the terraform cli. (eg: `./bin/terraform plan` will execute Terraform's "plan" command).
```

#### Terraform state management

> Terraform manage a [state](https://www.terraform.io/docs/state/index.html) of your infrastructure. By default this state is stored locally on your machine but it is highly recommanded to use a [remote backend](https://www.terraform.io/docs/state/remote.html).

Reviewer: ```suggestion
Terraform manages a [state](https://www.terraform.io/docs/state/index.html) of your infrastructure. By default this state is stored locally on your machine but it is highly recommanded to use a [remote backend](https://www.terraform.io/docs/state/remote.html).
```

> In a first time, the state management was included in Marsha but doing this you can't choose how you manage your state and it remains difficult for us to make state operation. We choose to not commit the state management in Marsha, you have to do it by yourself. 

Reviewer: I think it doesn't help and complicates understanding things
```suggestion
```

> You will find all you nedd to configure a remote backend in the Terraform documentation: https://www.terraform.io/docs/configuration/blocks/backends/index.html

Reviewer: ```suggestion
You will find all you need to configure a remote backend in the Terraform documentation: https://www.terraform.io/docs/configuration/blocks/backends/index.html
```

> ⚠ You must configure your state management before running any command after. The first `init` will initiate your state and after that you will have to deal with state migration if you want to modify it. you can create a file `src/aws/state.tf` and `src/aws/shared-resources/state.tf` to configure a backend, there is an example in each project (`state.tf.dist` file).

Reviewer: ```suggestion
⚠ You must configure your state management before running any of the commands hereafter. The first `init` will initiate your state and after that you will have to deal with state migration if you want to modify it. You can create a file `src/aws/state.tf` and `src/aws/shared-resources/state.tf` to configure a backend, there is an example in each project (`state.tf.dist` file).
```

🔧 **Before you go further**, you need to create `./src/aws/env.d/development` and replace the relevant values with your own. You can take a look at the [environment documentation](https://github.com/openfun/marsha/blob/master/docs/env.md#2-environment-to-deploy-marsha-to-aws) for more details on this topic. You should use this command to create the file from the existing model:

>     $ cp ./src/aws/env.d/development.dist ./src/aws/env.d/development

Author: I added line 118: 

```
$ cd src/aws
$ make init
```

WDYT ?

>     $ cp ./src/aws/env.d/development.dist ./src/aws/env.d/development

Reviewer 2: :+1:

All tasks related to this environment are run from the `./src/aws` directory. We use `Terraform` to keep this infrastructure configuration as code and easily manage several independent deployments of the whole `AWS` infrastructure.

There are 2 Terraform projects in Marsha with two different purposes:

- `./src/aws/shared_resources`: this project manages resources common to all marsha environments on the same AWS account. These resources must not live in different workspaces so you must work in the `default` workspace. To ease the use of this project, a dedicated script is available in `./src/aws/bin/shared-resources` which uses and configures the `Terraform` docker image. You have to run a Terraform command as if you were using the terraform cli. (eg: `./bin/shared-resources plan` will execute Terraform's "plan" command).

> - `./src/aws`: this is the main project we use, most of the infrastructure is managed here (in all `*.tf` files). This project must use [`Terraform` workspaces](https://www.terraform.io/docs/state/workspaces.html) and we highly recommand you to not use the default one. With multiple workspaces you can manage multiple instances of Marsha with one AWS account. To ease the use of this project, a dedicated script is available in `./src/aws/bin/terraform` which uses and configures the `Terraform` docker image. You have to run a Terraform command as if you were using the terraform cli. (eg: `./bin/terraform plan` will execute Terraform's "plan" command).

Reviewer: ```suggestion
- `./src/aws`: this is the main project we use, most of the infrastructure is managed here (in all `*.tf` files). This project must use [`Terraform` workspaces](https://www.terraform.io/docs/state/workspaces.html) and we highly recommand you to not use the default one. With multiple workspaces, you can manage multiple environments for your Marsha instance with a single AWS account. To ease the use of this project, a dedicated script is available in `./src/aws/bin/terraform` which uses and configures the `Terraform` docker image. You have to run a Terraform command as if you were using the terraform cli. (eg: `./bin/terraform plan` will execute Terraform's "plan" command).
```
